File: src/com/company/BitQustionNaiveSolutions/Power_Of_Two.java
package com.company.BitQustionNaiveSolutions;

public class Power_Of_Two {
    public static void main(String [] args){

        System.out.println(powOfTwo(16));

    }
    static Boolean powOfTwo(int n){

        if(n==1){
            return false;
        }
        while (n!=1){
            if (n%2!=0){
                return false;
            }
            n=n/2;
        }
        return true;
    }

}

File: src/com/company/BitManipulation/powerOftwo.kt
package com.company.BitManipulation

fun main(){

    powerOftwo(0)
}

fun powerOftwo(k:Int) {

    var n=k
    var count=0;
       while (n>0){
           n=n and (n-1)
           count++;
    }
    if (count==1){
        print("Yes")
    }else{
        print("No")
    }

}